### DISCUSSION TITLE
Rimozione entità dopo il riavvio

### DISCUSSION THREAD

**Discussion by FedericoPaiano** [indomus/forum#1170] [HUB: Home Assistant]
Salve a tutti e spero di poter risolvere il problema tramite questa comunità.
Sto lavorando ad un progetto di domotica che prevede la creazione di una schermata dove inserire le entità da inserire in vari gruppi.
I gruppi sono creati tramite il linguaggio yaml e le entità vengono inserite attraverso i servizi "group.set" e "group.add_entities".
I servizi funzionano bene ma al riavvio del sistema operativo i gruppi creati ritornano ad essere vuoti, così come creati.
Ho effettuato varie ricerche senza risultato e chiedo al gruppo se c'è un sistema che mi permetta di far memorizzare permanentemente ad Home Assistance le entità inserite anche dopo il riavvio, ovviamente senza inserirle manualmente tramite il linguaggio yaml.
Sperando di essere stato chiaro, ringrazio tutti.

**Comment by snakuzzo** [comment]
Ciao...incuriosito, ho fatto una prova proprio ora ed in effetti il gruppo non viene salvato.
Non so se sia un bug o una cosa voluta...ma potresti risolvere facendo una cosa del tipo:

- Salvi le entità che aggiungi in una `input_select` o in un `input_select`
- Fai un'automazione che all'avvio di HA recupera quelle entità e crea un gruppo

**Reply by FedericoPaiano** [reply]
Ciao Snakuzzo,
Grazie della cortese risposta. Credo che sia un limite di home assistant dovuto al fatto che quando il sistema viene riavviato si va a rileggere i gruppi senza entità così come li ho configurati nel file .yaml che ho costruito.
Per gli input il valore viene memorizzato ma per i gruppi credo sia difficile implementarlo, posto che in informatica nulla è impossibile.
Purtroppo non conosco nessun plugin che può risolvere il problema.
Così come hai scritto, ho già costruito degli script che si avviano all'avvio del sistema, ma la soluzione non è ciò che volevo, poiché devo comunque riempire i gruppi con degli input che, anche se non dovessero essere riempiti con le eventuali informazioni, si devono comunque precaricare; avrei preferito invece un comportamento più "dinamico".
E' difficile da spiegare, ma credo che grosso modo ci siamo capiti.
Grazie ancora dell'interessamento, Federico.

**Reply by snakuzzo** [reply]
Alternativa...ti fai uno script shell o python (esterno...non python script di HA) e ti crei dinamicamente il file yaml di quel gruppo.
Io faccio una cosa simile per generare il package per skyq con tutta la lista canali aggiornata.

**Reply by FedericoPaiano** [reply]
Dalla tua risposta capisco che sei un programmatore, ma io non ho le basi.
Gradirei imparare a programmare e capire come posso integrare python ad home assistant.
Grazie ancora.

**Reply by snakuzzo** [reply]
Programmatore è un parolone...direi più smanettone :D
Di fatto HA ti offre già la possibilità di fare script python ([QUI]https://indomus.it/componenti/componente-home-assistant-python-scripts/) nostro articolo)...ma comunque non potresti fare questa cosa perché sei limitato non potendo importare librerie diverse da quelle già a disposizione. Potresti usare [appdaemon](https://appdaemon.readthedocs.io), ma sinceramente, anche se non l'ho mai usato, lo trovo un po' inutile per questo scopo.
Io, come già detto prima, lato HA farei uno [shell command](https://www.home-assistant.io/integrations/shell_command/) che richiama uno script bash o uno script python esterno.

**Reply by FedericoPaiano** [reply]
AppDaemon infatti é per creare automazioni.
La ShellCommand non fa al caso mio perché voglio creare un'interfaccia utente che permetta agli utilizzatori di creare gruppi senza andare in yaml.
Vediamo cosa posso ricavare dall'articolo che mi hai indicato. Comunque a modo mio ho risolto creando circa 500 input_text (per ora). Quando il progetto sarà completo lo pubblicherò per permettere a chi é più esperto di me di migliorarlo.
